cpp/tests/jit/jit-cache-test.hpp

```diff
@@ -107,10 +107,10 @@ struct JitCacheTest : public ::testing::Test, public cudf::jit::cudfJitCache {
 
 /**
  * @brief Similar to JitCacheTest but it doesn't run warmUp() test in SetUp and
- * purgeFileCache() in TearDown
+ * purgeFileCache() in SetUp and TearDown
  */
 struct JitCacheMultiProcessTest : public JitCacheTest {
-  virtual void SetUp() { purgeFileCache(); }
```

Comment:
Just a question. Does this test the new cache path logic or is this fixing something else?

Also, after this is merged, this will create a new cache directory -- any old cache directory remains unless I delete it first, right?

Reply:
> Does this test the new cache path logic or is this fixing something else?

Details in https://github.com/rapidsai/cudf/pull/7090/commits/742c0b430d75f6e8e574c45f81fe654bbc9bd451 message. Basically, the test forks before Cuda context can be initialised. Now, getting the cache directory would itself cause the context to be initialised. Trying to avoid that in the setup.

> Also, after this is merged, this will create a new cache directory -- any old cache directory remains unless I delete it first, right?

This is true since we moved the cache directory from `/tmp` to `$HOME`. Cache is going to keep growing until we decide to add cleanup mechanisms, which considering the size of the caches, isn't a priority.